File: eAutoShop/eAutoShop.Model/SearchObjects/OrderSearchObject.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace eAutoShop.Model.SearchObjects
{
    public class OrderSearchObject : BaseSearchObject
    {
        public string? CustomerName { get; set; }
        public string? State { get; set; }

        public double? MinTotalAmount { get; set; }
        public double? MaxTotalAmount { get; set; }

        public DateTime? MinOrderDate { get; set; }
        public DateTime? MaxOrderDate { get; set; }
        public DateTime? MinShippingDate { get; set; }
        public DateTime? MaxShippingDate { get; set; }
        public bool? HasDiscount { get; set; } 
        public bool? IncludeItems { get; set; } 
    }

}

File: eAutoShop/eAutoShop.UI/eautoshop_mobile/lib/models/order/order_search_object.dart
import 'package:json_annotation/json_annotation.dart';

part 'order_search_object.g.dart';

@JsonSerializable(includeIfNull: false)
class OrderSearchObject {
  int? page;
  int? pageSize;
  String? customerName;
  String? state;
  double? minTotalAmount;
  double? maxTotalAmount;
  DateTime? minOrderDate;
  DateTime? maxOrderDate;
  DateTime? minShippingDate;
  DateTime? maxShippingDate;
  bool? hasDiscount;
  bool? includeItems;

  OrderSearchObject({
    this.page,
    this.pageSize,
    this.customerName,
    this.state,
    this.minTotalAmount,
    this.maxTotalAmount,
    this.minOrderDate,
    this.maxOrderDate,
    this.minShippingDate,
    this.maxShippingDate,
    this.hasDiscount,
    this.includeItems,
  });

  factory OrderSearchObject.fromJson(Map<String, dynamic> json) =>
      _$OrderSearchObjectFromJson(json);

  Map<String, dynamic> toJson() => _$OrderSearchObjectToJson(this);
}
